src/io/SystemMetadata.tsx

```diff
@@ -0,0 +1,38 @@
+export interface SystemMetadata {
+    cpuUsage: number;
+    memoryUsagePercentage: number;
+    memoryUsed: number;
+    memoryTotal: number;
+    memoryAvailable: number;
+    systemUptime: number;
+}
+
+export class SystemMetadataUtils {
+
+    public static memoryStringFormat(numberToFormat: number) {
+        // number of bytes in a terabyte, gigabyte, and megabyte.
+        let result: string;
+        if (numberToFormat >= 1099511627776) {
+            result = (numberToFormat / 1099511627776).toPrecision(4).toString()  + "TB";
```

Comment:
Here comes one of the most confusing stuff in everything computer related...
I'm quite sure that what you are actually computing are not `TB`s but `TiB`s (same for the other units).
Bytes multiples are sometimes expressed with powers of 10 (International System) and sometimes with powers of 2.
Basically it works like this:
* 1000 (10^3) bytes = 1kB (Kilobyte)
* 1024 (2^10) bytes = 1KiB (Kibibyte)
Same for MBs, GBs, etc.
See [here](https://en.wikipedia.org/wiki/Kibibyte) (table on the right) and [here](https://blog.forret.com/2005/02/04/binary-confusion-kilobytes-and-kibibytes/) for more

So I'd suggest you pick if you want to use base-10 or base-2 and fix the code accordingly.